File: backend/app/services/dashboard_service.py
from sqlalchemy.orm import Session
from app.models.user import User
from app.models.species import Species
from app.models.protected_area import ProtectedArea
from app.models.wildlife_observation import WildlifeObservation


def get_dashboard_stats(db: Session):
    return {
        "total_users": db.query(User).count(),
        "total_species": db.query(Species).count(),
        "total_protected_areas": db.query(ProtectedArea).count(),
        "total_observations": db.query(WildlifeObservation).count(),
    }

File: frontend/src/services/dashboardService.ts
import api from "./api";

export interface DashboardStats {
  total_users: number;
  total_species: number;
  total_protected_areas: number;
  total_observations: number;
}

export async function getDashboardStats() {
  const response = await api.get<DashboardStats>(
    "/dashboard/stats"
  );

  return response.data;
}